contrib/zpages/src/main/java/io/opencensus/contrib/zpages/RpczZPageHandler.java

```diff
@@ -0,0 +1,454 @@
+        || view == RPC_SERVER_ERROR_COUNT_MINUTE_VIEW) {
+      snapshot.errorsMinute = ((MeanData) data).getCount();
+    } else if (view == RPC_CLIENT_ERROR_COUNT_HOUR_VIEW
+        || view == RPC_SERVER_ERROR_COUNT_HOUR_VIEW) {
+      snapshot.errorsHour = ((MeanData) data).getCount();
+    } else if (view == RPC_CLIENT_REQUEST_BYTES_VIEW || view == RPC_SERVER_REQUEST_BYTES_VIEW) {
+      DistributionData distributionData = (DistributionData) data;
+      snapshot.inputRateTotal =
+          distributionData.getCount()
+              * distributionData.getMean()
+              / BYTES_PER_KB
+              / getDurationInSecs((CumulativeData) windowData);
+    } else if (view == RPC_CLIENT_REQUEST_BYTES_MINUTE_VIEW
+        || view == RPC_SERVER_REQUEST_BYTES_MINUTE_VIEW) {
+      MeanData meanData = (MeanData) data;
+      snapshot.inputRateMinute =
+          meanData.getMean() * meanData.getCount() / BYTES_PER_KB / SECONDS_PER_MINUTE;
+    } else if (view == RPC_CLIENT_REQUEST_BYTES_HOUR_VIEW
+        || view == RPC_SERVER_REQUEST_BYTES_HOUR_VIEW) {
+      MeanData meanData = (MeanData) data;
+      snapshot.inputRateHour =
+          meanData.getMean() * meanData.getCount() / BYTES_PER_KB / SECONDS_PER_HOUR;
+    } else if (view == RPC_CLIENT_RESPONSE_BYTES_VIEW || view == RPC_SERVER_RESPONSE_BYTES_VIEW) {
+      DistributionData distributionData = (DistributionData) data;
+      snapshot.outputRateTotal =
+          distributionData.getCount()
+              * distributionData.getMean()
+              / BYTES_PER_KB
+              / getDurationInSecs((CumulativeData) windowData);
+    } else if (view == RPC_CLIENT_RESPONSE_BYTES_MINUTE_VIEW
+        || view == RPC_SERVER_RESPONSE_BYTES_MINUTE_VIEW) {
+      MeanData meanData = (MeanData) data;
+      snapshot.outputRateMinute =
+          meanData.getMean() * meanData.getCount() / BYTES_PER_KB / SECONDS_PER_MINUTE;
+    } else if (view == RPC_CLIENT_RESPONSE_BYTES_HOUR_VIEW
+        || view == RPC_SERVER_RESPONSE_BYTES_HOUR_VIEW) {
+      MeanData meanData = (MeanData) data;
+      snapshot.outputRateHour =
+          meanData.getMean() * meanData.getCount() / BYTES_PER_KB / SECONDS_PER_HOUR;
+    } else if (view == RPC_CLIENT_STARTED_COUNT_MINUTE_VIEW
+        || view == RPC_SERVER_STARTED_COUNT_MINUTE_VIEW) {
+      snapshot.countMinute = ((MeanData) data).getCount();
+    } else if (view == RPC_CLIENT_STARTED_COUNT_HOUR_VIEW
+        || view == RPC_SERVER_STARTED_COUNT_HOUR_VIEW) {
+      snapshot.countHour = ((MeanData) data).getCount();
+    }
+  }
+
+  // Calculates the duration of the given CumulativeData in seconds.
+  private static double getDurationInSecs(CumulativeData cumulativeData) {
+    return toSecs(cumulativeData.getEnd().subtractTimestamp(cumulativeData.getStart()));
+  }
+
+  // Converts a Duration to seconds.
+  private static double toSecs(Duration duration) {
+    return duration.getNanos() / NANOS_PER_SECOND + duration.getSeconds();
```

Comment:
Based on the definition of the getNanos this will always be equivalent to:

return duration.getSeconds();

Am I missing something?

Reply:
I thought nanoseconds would be kept separately?

IIUC for example, consider 
`Duration duration = Duration.create(/* seconds */ 5, /* nanos */ 5 * 1e8);`. 
If we want to convert `duration` to seconds, it should be 5.5 seconds, but `duration.getSeconds()` will only return 5 seconds. Is that right?

Reply:
Please explain that in the comment. Maybe also rename toDoubleSeconds.

Reply:
Sure, I added some comments on this.